### PR TITLE
Query keys refactor

### PR DESCRIPTION
**Pull request by LeonmanRolls**
fixes fet-1114

### CODE REVIEW

File: src/transaction-flow/input/ProfileEditor/ProfileEditor.test.tsx [hunk 1]
```diff
@@ -247,12 +249,14 @@ describe('ProfileEditor with old resolver', () => {
...
     mockUseBasicName.mockReturnValue({ isWrapped: false })
+    mockUseAccount.mockReturnValue({ address: '0x123' })
   })
 
   it('should submit to key value to alternative dispatch if resolver address is not current', async () => {
-    render(
+    const { debug } = render(
       <ProfileEditor data={{ name: 'test.eth' }} dispatch={mockDispatch} onDismiss={() => {}} />,
     )
+    debug()
```
Comment (TateB) [inline src/transaction-flow/input/ProfileEditor/ProfileEditor.test.tsx:259]: remove debug

File: src/utils/cacheKeyFactory.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import type { TransactionItem } from '@app/transaction-flow/transaction'
+
+export const useQueryKeys = () => {
+  const { address } = useAccount()
+  const chainId = useChainId()
+
+  const globalKeys = [chainId, address]
+
+  return {
+    dogfood: (inputString?: string) => [...globalKeys, 'getAddr', inputString],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:14]: doesn't look like this is being used

File: src/utils/cacheKeyFactory.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    transactionStageModal: {
+      prepareTransaction: (
+        uniqueTransactionIdentifiers: ReturnType<typeof uniqueTransactionIdentifierGenerator>,
+      ) => [...globalKeys, 'prepareTx', uniqueTransactionIdentifiers],
+      transactionError: (transactionHash?: string) => [...globalKeys, 'txError', transactionHash],
+    },
+    claimDomain: (name: string, syncWarning: boolean) => [
+      ...globalKeys,
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:22]: this doesn't need global keys

File: src/utils/cacheKeyFactory.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+      'currentExternalTransactionId',
+      currentExternalTransactionId,
+    ],
+    avatar: {
+      avatar: (network: number, name: string | null | undefined) => [
+        ...globalKeys,
+        'getAvatar',
+        name,
+        network,
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:38]: network isnt needed since there is global chainid

File: src/utils/cacheKeyFactory.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    ],
+    beautifiedName: (name: string) => [...globalKeys, 'beautify', name],
+    blockTimestamp: [...globalKeys, 'use-block-timestamp'],
+    currentBlockTimestamp: [...globalKeys, 'currentBlockTimestamp'],
+    decryptName: (name: string) => [...globalKeys, 'graph', 'decryptName', name],
+    getDNSOwner: (name: string) => [...globalKeys, 'getDNSOwner', name],
+    getDNSProof: (name: string) => [...globalKeys, 'getDNSProof', name],
+    estimateGasLimitForTransactions: (transactions: TransactionItem[], extraKeys: string[]) => [
+      ...globalKeys,
+      'use-estimate-gas-limit-for-transactions',
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:58]: can this be camel case

File: src/utils/cacheKeyFactory.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+      ...transactions,
+      ...extraKeys,
+    ],
+    estimateRegistration: [...globalKeys, 'gas-costs'],
+    ethPrice: [...globalKeys, 'use-eth-price'],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:63]: camelCase

File: src/utils/cacheKeyFactory.ts [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    decryptName: (name: string) => [...globalKeys, 'graph', 'decryptName', name],
+    getDNSOwner: (name: string) => [...globalKeys, 'getDNSOwner', name],
+    getDNSProof: (name: string) => [...globalKeys, 'getDNSProof', name],
+    estimateGasLimitForTransactions: (transactions: TransactionItem[], extraKeys: string[]) => [
+      ...globalKeys,
+      'use-estimate-gas-limit-for-transactions',
+      ...transactions,
+      ...extraKeys,
+    ],
+    estimateRegistration: [...globalKeys, 'gas-costs'],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:62]: camelCase

File: src/utils/cacheKeyFactory.ts [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    },
+    basicName: (normalisedName: string) => [
+      ...globalKeys,
+      'batch',
+      'getOwner',
+      'getExpiry',
+      normalisedName,
+    ],
+    beautifiedName: (name: string) => [...globalKeys, 'beautify', name],
+    blockTimestamp: [...globalKeys, 'use-block-timestamp'],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:51]: camelCase

File: src/utils/cacheKeyFactory.ts [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+    estimateRegistration: [...globalKeys, 'gas-costs'],
+    ethPrice: [...globalKeys, 'use-eth-price'],
+    exists: (name: string) => [...globalKeys, 'getOwner', name],
+    expiry: (name: string) => [...globalKeys, 'useExpiry', name],
+    faucet: (localAddress?: string) => [...globalKeys, 'getFaucetEligibility', localAddress],
+    getABI: (name: string) => [...globalKeys, 'useGetABI', name],
+    getHistory: (name: string) => [...globalKeys, 'graph', 'getHistory', name],
+    getWrapperData: (name: string) => [...globalKeys, 'getWrapperData', name],
+    hasSubnames: (name: string) => [...globalKeys, 'graph', 'getSubnames', name],
+    namesFromAddress: (localAddress?: string) => [...globalKeys, 'graph', 'names', localAddress],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:71]:
```suggestion
    namesFromAddress: (localAddress?: string) => [...globalKeys, 'graph', 'getNames', localAddress],
```

File: src/utils/cacheKeyFactory.ts [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+      'use-estimate-gas-limit-for-transactions',
+      ...transactions,
+      ...extraKeys,
+    ],
+    estimateRegistration: [...globalKeys, 'gas-costs'],
+    ethPrice: [...globalKeys, 'use-eth-price'],
+    exists: (name: string) => [...globalKeys, 'getOwner', name],
+    expiry: (name: string) => [...globalKeys, 'useExpiry', name],
+    faucet: (localAddress?: string) => [...globalKeys, 'getFaucetEligibility', localAddress],
+    getABI: (name: string) => [...globalKeys, 'useGetABI', name],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:67]:
```suggestion
    getABI: (name: string) => [...globalKeys, 'getABI', name],
```

File: src/utils/cacheKeyFactory.ts [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+    getHistory: (name: string) => [...globalKeys, 'graph', 'getHistory', name],
+    getWrapperData: (name: string) => [...globalKeys, 'getWrapperData', name],
+    hasSubnames: (name: string) => [...globalKeys, 'graph', 'getSubnames', name],
+    namesFromAddress: (localAddress?: string) => [...globalKeys, 'graph', 'names', localAddress],
+    price: (type: 'legacy' | 'new', names: string[]) => [...globalKeys, 'usePrice', type, ...names],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:72]: should be getPrice

File: src/utils/cacheKeyFactory.ts [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+    primary: (localAddress: string) => [...globalKeys, 'getName', localAddress],
+    profile: (name: string) => [...globalKeys, 'garph', 'getProfile', name],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:74]: graph misspelled

File: src/utils/cacheKeyFactory.ts [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+    registrationDate: (name: string) => [...globalKeys, 'garph', 'getRegistrationDate', name],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:75]: graph misspelled

File: src/utils/cacheKeyFactory.ts [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+    resolver: (name: string) => [...globalKeys, 'use-resolver', name],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:76]: should be getResolver

File: src/utils/cacheKeyFactory.ts [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+    resolverExists: (name: string) => [...globalKeys, 'graph', 'getResolverExists', name],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:77]: has getResolver dependency

File: src/utils/cacheKeyFactory.ts [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+    resolverHasInterfaces: (interfaceNames: string, resolverAddress?: string) => [
+      ...globalKeys,
+      'validateResolver',
+      resolverAddress,
+      interfaceNames,
+      chainId,
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:83]: chainId already in globalkeys

File: src/utils/cacheKeyFactory.ts [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+    ],
+    resolverStatus: (
+      name: string,
+      options: { skip?: boolean; skipCompare?: boolean },
+      profileResolverAddress?: string,
+    ) => [...globalKeys, 'resolverStatus', name, { profileResolverAddress, options }],
+    supportsTLD: (tld: string) => [...globalKeys, 'supportedTLD', tld],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:90]: fn name is supportsTLD

File: src/utils/cacheKeyFactory.ts [hunk 18]
```diff
@@ -0,0 +1,112 @@
...
+    validate: (input: string) => [...globalKeys, 'validate', input],
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:91]: this doesnt need global keys

File: src/utils/cacheKeyFactory.ts [hunk 19]
```diff
@@ -0,0 +1,112 @@
...
+    validateSubnameLabel: (validationName: string) => [
+      ...globalKeys,
+      'createSubname',
+      'getOwner',
+      validationName,
+    ],
+    wrapperApprovedForAll: (nameWrapperAddress: string, localAddress: string) => [
+      ...globalKeys,
+      'approvedForAll',
+      nameWrapperAddress,
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:101]: nw address is static for network

File: src/utils/cacheKeyFactory.ts [hunk 20]
```diff
@@ -0,0 +1,112 @@
...
+      localAddress,
+    ],
+    zorb: (input: string, type: string, bg: string, fg: string, accent: string) => [
+      ...globalKeys,
```
Comment (TateB) [inline src/utils/cacheKeyFactory.ts:105]: this doesnt need global keys

**Review by TateB — CHANGES_REQUESTED** [review]
ik this is just a direct port of query keys into fn format but i think its a good time to make all of the keys consistent so i left comments for existing mistakes/inconsistencies

File: src/hooks/useEstimateRegistration.ts [hunk 21]
```diff
@@ -31,14 +32,17 @@ const useEstimateRegistration = (
...
-  const { data: gasCosts, isLoading: gasCostsLoading } = useQuery(['gas-costs'], async () => {
-    const addr = (await import('@app/assets/gas-costs/addr.json'))
-      .default as unknown as GasCostData[]
-    const text = (await import('@app/assets/gas-costs/text.json'))
-      .default as unknown as GasCostData[]
-
-    return { addr, text }
-  })
+  const { data: gasCosts, isLoading: gasCostsLoading } = useQuery(
+    useQueryKeys().estimateRegistration,
```
[the remark this answers is not in the document]
Comment (LeonmanRolls) [reply src/hooks/useEstimateRegistration.ts:36]: it's possible goerli or another chain could diverge in the future

File: src/utils/cacheKeyFactory.ts (hunk 3 above)
Comment (LeonmanRolls) [reply src/utils/cacheKeyFactory.ts:22]: I'd prefer to just have the global keys actually be global

Comment (LeonmanRolls) [reply src/utils/cacheKeyFactory.ts:22]: changed my mind, put them under their own key

**Pull request merged**